### PR TITLE
feat(2418): Modify Badge API

### PR DESCRIPTION
**Pull request by mchengat**
## Context

Badge api uses shields.io url to create badges. 

## Objective

Modify badge api to use npm library

## References
Related to https://github.com/screwdriver-cd/screwdriver/issues/2418

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/pipelines/badge.js [hunk 1]
```diff
@@ -90,28 +87,31 @@ module.exports = config => ({
...
+
             const { statusColor } = config;
             const badgeConfig = {
-                badgeService,
-                statusColor,
-                encodeBadgeSubject
+                statusColor
+            };
+
+            const getBadge = badgeObject => {
```
Comment (jithine) [inline plugins/pipelines/badge.js:96]: why an inline function? can you move it to top level where `getLabels` is ?

File: plugins/pipelines/jobBadge.js [hunk 2]
```diff
@@ -47,13 +45,15 @@ module.exports = config => ({
...
-            const { encodeBadgeSubject } = request.server.plugins.pipelines;
             const { statusColor } = config;
             const badgeConfig = {
-                badgeService,
-                statusColor,
-                encodeBadgeSubject
+                statusColor
+            };
+
+            const getBadge = badgeObject => {
```
Comment (jithine) [inline plugins/pipelines/jobBadge.js:53]: move this function out of the handler function

Comment (mchengat) [reply plugins/pipelines/jobBadge.js]: Updated

File: plugins/pipelines/badge.js [hunk 3]
```diff
@@ -3,20 +3,18 @@
...
+ * @method getBadge
  * @param  {Object} statusColor             Mapping for status and color
- * @param  {Function} encodeBadgeSubject    Function to encode subject
  * @param  {Array}  [buildsStatus=[]]       An array of builds
- * @param  {String} [subject='job']         Subject of the badge
+ * @param  {String} [label='pipeline']         Subject of the badge
  * @return {String}
  */
-function getUrl({ badgeService, statusColor, encodeBadgeSubject, buildsStatus = [], subject = 'pipeline' }) {
+function getBadge({ statusColor, buildsStatus = [], label = 'pipeline' }) {
```
Comment (tkyi) [inline plugins/pipelines/badge.js:17]: would it be better to move this function to a helper file like this: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/helper.js?

**Review by tkyi — COMMENTED** [review]
Might be better to add `getPipelineBadge` and `getJobBadge` helper functions to `helper.js` vs add to new file `getBadge.js`. The rest of the files in that folder are all endpoints.

File: plugins/pipelines/helper.js [hunk 4]
```diff
@@ -48,7 +49,68 @@ const sanitizeRootDir = (rootDir = '') => {
...
 
+/**
+ * Generate Badge for pipeline
+ * @method getPipelineBadge
+ * @param  {Object} statusColor             Mapping for status and color
+ * @param  {Array}  [buildsStatus=[]]       An array of builds
+ * @param  {String} [label='pipeline']         Subject of the badge
+ * @return {String}
+ */
+// function getPipelineBadge({ statusColor, buildsStatus = [], label = 'pipeline' }) {
```
Comment (jithine) [inline plugins/pipelines/helper.js:60]: remove commented line ?

File: plugins/pipelines/helper.js [hunk 5]
```diff
@@ -48,7 +49,68 @@ const sanitizeRootDir = (rootDir = '') => {
...
+
+/**
+ * Generate Badge for Job
+ * @method getJobBadge
+ * @param  {Object} statusColor             Mapping for status and color
+ * @param  {Array}  [builds=[]]       An array of builds
+ * @param  {String} [label='job']         Subject of the badge
+ * @return {String}
+ */
+// function getJobBadge({ statusColor, builds = [], label = 'job' }) {
```
Comment (jithine) [inline plugins/pipelines/helper.js:94]: remove commented line ?

**Review by tkyi — APPROVED** [review]

**Pull request merged**